apis/apps/v1alpha1/componentdefinition_types.go

```diff
@@ -190,6 +190,13 @@ type ComponentDefinitionSpec struct {
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
 
+	// Defines static annotations that will be patched to all k8s resources created for the component.
+	// If a annotation key conflicts with any other system annotations or user-specified annotations, it will be silently ignored.
+	// This field is immutable.
```

Comment:
then enforce it by a validation rule?

Reply:
> then enforce it by a validation rule?

This is to keep consistency with other fields, and validation will be added once the API is finally stable.